File: employee-portal/src/lib/businessModules/dental/api/models/ChildSearchResult.ts
/**
 * Copyright 2025 cronn GmbH
 * SPDX-License-Identifier: AGPL-3.0-only
 */

import { ApiChildSearchResult, ApiGender } from "@eshg/dental-api";

export interface ChildSearchResult {
  readonly id: string;
  readonly firstName: string;
  readonly lastName: string;
  readonly dateOfBirth: Date;
  readonly groupName: string;
  readonly gender?: ApiGender;
}

export function mapChildSearchResult(
  response: ApiChildSearchResult,
): ChildSearchResult {
  return {
    ...response,
  };
}

File: backend/dental/src/main/java/de/eshg/dental/api/ChildSearchResult.java
/*
 * Copyright 2025 cronn GmbH
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package de.eshg.dental.api;

import de.eshg.base.GenderDto;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.UUID;

public record ChildSearchResult(
    @NotNull UUID id,
    @NotNull String firstName,
    @NotNull String lastName,
    @NotNull LocalDate dateOfBirth,
    String groupName,
    GenderDto gender) {}
